pyuvdata/uvdata/uvh5.py

```diff
@@ -136,6 +131,224 @@ def _write_complex_astype(data, dset, indices):
     return
 
 
+def _convert_to_slices(indices, max_nslice_frac=0.1):
```

Comment:
That's right. The idea here is that an indexing list, say, like `indices = [0, 1, 2, 3, 10, 11, 12, 13]` can be represented as `[slice(0, 4), slice(10, 14)]`, and performing `dset[slice(0, 4)]` and then `dset[slice(10, 14)]` in serial is faster than `dset[indices]`. This is still multi-slicing, but its still better than direct fancy indexing.

However, say we have an indices list that is not really representable in terms of slices (take for example a list of random integers) `[0, 10, 25, 55, 105, 235]`, then this could be represented as `[slice(0, 11, 10), slice(25, 56, 30), slice(105, 236, 130)]`, but this begins to be suboptimal as the number of elements increases. In the limit that we have Nslices = Nindices then the slice indexing is actually slower than the fancy indexing, so its best to just keep `indices` as is. That's what this parameter is trying to get at.